ksql-common/src/main/java/io/confluent/ksql/util/ReservedInternalTopics.java

```diff
@@ -0,0 +1,129 @@
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @return The command topic name.
+   */
+  public static String commandTopic(final KsqlConfig ksqlConfig) {
+    return toKsqlInternalTopic(ksqlConfig, KSQL_COMMAND_TOPIC_SUFFIX);
+  }
+
+  /**
+   * Returns the internal KSQL configs topic (used for KSQL standalone)
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @return The configurations topic name.
+   */
+  public static String configsTopic(final KsqlConfig ksqlConfig) {
+    return toKsqlInternalTopic(ksqlConfig, KSQL_CONFIGS_TOPIC_SUFFIX);
+  }
+
+  /**
+   * Returns the KSQL processing log topic.
+   * <p/>
+   * This is not an internal topic in the sense that users are intentionally meant to read from
+   * this topic to identify deserialization and other processing errors, define a KSQL stream on
+   * it, and potentially issue queries to filter from it, etc. This is why it is not prefixed in
+   * the way KSQL internal topics are.
+   *
+   * @param config The Processing log config, which is used to extract the processing topic suffix
+   * @param ksqlConfig The KSQL config, which is used to extract the KSQL service id.
+   * @return The processing log topic name.
+   */
+  public static String processingLogTopic(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig
+  ) {
+    final String topicNameConfig = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    if (topicNameConfig.equals(ProcessingLogConfig.TOPIC_NAME_NOT_SET)) {
+      return String.format(
+          "%s%s",
+          ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+          ProcessingLogConfig.TOPIC_NAME_DEFAULT_SUFFIX
+      );
+    } else {
+      return topicNameConfig;
+    }
+  }
+
+  /**
+   * Compute a name for a KSQL internal topic.
+   *
+   * @param ksqlConfig The KSQL config, which is used to extract the internal topic prefix.
+   * @param topicSuffix A suffix that is appended to the topic name.
+   * @return The computed topic name.
+   */
+  private static String toKsqlInternalTopic(final KsqlConfig ksqlConfig, final String topicSuffix) {
+    return String.format(
+        "%s%s_%s",
+        KSQL_INTERNAL_TOPIC_PREFIX,
+        ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+        topicSuffix
+    );
+  }
+
+  private final List<Pattern> systemInternalTopics;
+
+  public ReservedInternalTopics(final KsqlConfig ksqlConfig) {
+    try {
+      this.systemInternalTopics = ksqlConfig.getList(KsqlConfig.SYSTEM_INTERNAL_TOPICS_CONFIG)
+          .stream()
+          .map(Pattern::compile)
+          .collect(Collectors.toList());
+    } catch (final Exception e) {
+      final String message = "Cannot get a list of system internal topics due to an invalid " +
+          "configuration in '" + KsqlConfig.SYSTEM_INTERNAL_TOPICS_CONFIG + "'";
+
+      LOG.error(message + ": " + e.getMessage());
```

Comment:
I thought that initially, but while I was running some tests, I found the exception was only displayed on the client side. The server log didn't have any error. Being this a server configuration, I wanted to warn the KSQL server admin about this error too so they can fix it instead of relying on the client to send the error to the server admin.

Reply:
I didn't change this in the updated PR because of above reasons. Let me know what you think.

Reply:
Log and throw really is a anti pattern. Better to work out why the exception was not being logged on startup, as that omission could be causing other failures to not be logged, i.e. its a bug that needs fixing.